example/service/s3/listObjects/listObjects.go

```diff
@@ -0,0 +1,129 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+)
+
+var (
+	bucketName      string
+	objectPrefix    string
+	objectDelimiter string
+)
+
+func init() {
+	flag.StringVar(&bucketName, "bucket", "", "The `name` of the S3 bucket to list objects from.")
+	flag.StringVar(&objectPrefix, "prefix", "", "The optional `object prefix` of the S3 Object keys to list.")
+	flag.StringVar(&objectDelimiter, "delimiter", "/",
+		"The optional `object key delimiter` used by S3 List objects to group object keys.")
+}
+
+// Lists all objects in a bucket using pagination
+//
+// Usage:
+// listObjects <bucket>
+func main() {
+	flag.Parse()
+	if len(bucketName) == 0 {
+		flag.PrintDefaults()
+		log.Fatalf("invalid parameters, bucket name required")
+	}
+
+	// Load the SDK's configuration from environment and shared config, and
+	// create the client with this.
+	cfg, err := config.LoadDefaultConfig()
+	if err != nil {
+		log.Fatalf("failed to load SDK configuration, %v", err)
+	}
+
+	client := s3.NewFromConfig(cfg)
+
+	// Set the parameters based on the CLI flag inputs.
+	params := &s3.ListObjectsV2Input{
+		Bucket: &bucketName,
+	}
+	if len(objectPrefix) != 0 {
+		params.Prefix = &objectPrefix
+	}
+	if len(objectDelimiter) != 0 {
+		params.Delimiter = &objectDelimiter
+	}
+
+	// TODO replace this with the code generate paginator when available
+	// s3.NewListObjectsV2Paginator()
+	p := NewS3ListObjectsV2Paginator(client, params)
+
+	// Iterate through the S3 object pages, printing each object returned.
+	var i int
+	log.Println("Objects:")
+	for p.HasMorePages() {
+		i++
+
+		page, err := p.NextPage(context.TODO())
```

Comment:
Nit: Can you add a comment on taking a new context each time, this avoids timeouts between page retrieval and seems good to call out.